draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1137,18 +1143,76 @@ SimulationMarker = {
 ~~~
 {: #simulation-marker-def title="SimulationMarker definition"}
 
-# Event definition guidelines
+# Event extensibility {#event-extensibility}
+
+The main qlog schema defined by this document describes log and trace formats
+that have related events. A minimal set of common events are presented in
+{{events}}. It is expected that logging is extended with specific, per-protocol
+event definitions that specify the name (category + type) and data needed for
+each individual event. Examples include the QUIC event definitions {{QLOG-QUIC}}
+and HTTP/3 event definitions {{QLOG-H3}}.
+
+New event definitions SHOULD follow the guidance in this section, which provides
+consistency that makes it easier for qlog implementers to support a wide variety
+of protocols.
+
+## Extended Event Schema
+
+New event definitions SHOULD be part of an extension schema, using the
+annotations and concepts presented in this section.
+
+An extension schema defines one or more qlog event categories that can contain
+multiple event definitions. Each category MUST have a globally unique category
+identifier. Each event MUST only belong to a single category. Per
+{{{#name-field}}}, event names are the concatenation of category and type.
+Therefore, these requirements ensure there are no ambiguities when multiple
+extension schema are in use.
+
+Implementations that might record events from extension schemas SHOULD list all
+category identifiers in use. This is achieved by including the appropriate URI
+in the `additional_event_schemas` field of the QlogFile ({{qlog-file-schema}})
+or QlogFileSeq ({{qlog-file-seq-schema}}). The `additional_event_schema` is a
+hint to tools about the possible events that a qlog file might contain. The file
+may contain events that do not belong to a listed category identifier. Tools
+MUST NOT treat this as an error; see {{tooling}}.
+
+Each extension schema is named by a URI. That URI MUST be absolute and MUST
+include a category identifier, indicated using a fragment identifier (characters
```

Comment:
This is inconsistent with how you defined a "schema" at the start. There you say:

> An extension schema defines one or more qlog event categories

and here 1 schema is named by 1 URI, but that URI then somehow contains a category identifier? So the URI doesn't identify the schema, it instead identifies a single category WITHIN an extension schema.

Reply:
agree, are you happier with something like 

```suggestion
Each category identifier in a schema is named by a URI. That URI MUST be absolute and MUST contain a fragment identifier (characters after a "#" in the URI) that indicates the category identifier. 
```

Reply:
yes, that's much more accurate :)